app/client/components/consent/PrivacySettings.tsx

```diff
@@ -212,10 +280,26 @@ export class PrivacySettings extends Component<{}, State> {
     );
   }
 
-  private buildState(iabVendorList: ParsedIabVendorList): Promise<void> {
+  private buildState(
+    guPurposeList: ParsedGuPurposeList,
+    iabVendorList: ParsedIabVendorList
+  ): Promise<void> {
+    // tslint:disable-next-line: no-object-mutation
+    this.guPurposeList = guPurposeList;
     // tslint:disable-next-line: no-object-mutation
     this.iabVendorList = iabVendorList;
 
+    const guState = readGuCookie();
+
+    let guPurposes = {};
```

Comment:
You could lose the `let` and `guState` here by doing:

```js
const guPurposes = readGuCookie() || guPurposeList.purposes.reduce((acc, purpose) => {
    return {
        ...acc,
        [purpose.id]: !purpose.id ? true : null
    };
}, {});
```

Reply:
Good point!